modules/custom/dkan_metastore/src/Service.php

```diff
@@ -254,7 +258,10 @@ public function put($schema_id, $identifier, string $data): array {
    */
   public function patch($schema_id, $identifier, $data) {
     $engine = $this->getEngine($schema_id);
-    if ($this->objectExists($schema_id, $identifier)) {
+    if ($existing_data = $this->objectExists($schema_id, $identifier)) {
```

Comment:
This code is exactly the same as the one in the put method. Should we move this "objectHasChanged" logic into its own method?

Reply:
Thanks, I will follow your suggestion:
- Restore the original `objectExists`
- Create a `objectHasChanged`